src/test/java/tests/FilterParserTests.java

```diff
@@ -72,6 +72,10 @@ public void unexpectedEOFs() {
             Parser.parse("a(b) ||");
             fail("Inputs which end unexpectedly should throw a parse exception");
         } catch (ParseException ignored) {}
+        try {
+            Parser.parse("a:b; c:d;e");
+            fail("Inputs which end unexpectedly should throw a parse exception");
```

Comment:
This isn't really a case of the input ending unexpectedly. Rather, it's because some kind of content is expected as the input to qualifier `a`, and `c:` (the start of a new qualifier) can't be taken as content.

It's a useful test, but we should make the assertion message more accurate and place it near the other related tests.